### PR TITLE
Add volunteer project routes

### LINKED ISSUE #38: Community businesses belongs to many funding programmes
User feedback via @SonjaTwine:

> On the "organisation" table, the value for "Funding programme" needs to be able to be more than 1 (i.e. a community business can be a member of several funding programmes)

### PR DESCRIPTION
**Pull request by eliascodes**
Fixes #38 

### Changes
* Added trigger on projects table
* Add projects methods to `VolunteerLogs` model
* Add projects CRUD endpoints

### CODE REVIEW

File: src/api/v1/community_businesses/volunteers/__tests__/projects.test.integration.ts [hunk 1]
```diff
@@ -0,0 +1,253 @@
+import * as Hapi from 'hapi';
+import * as Knex from 'knex';
+import { init } from '../../../../../server';
+import { getConfig } from '../../../../../../config';
+import { CommunityBusinesses, Organisation } from '../../../../../models';
+
+
+describe('API /community-businesses/{id}/volunteers', () => {
```
Comment (astroash) [inline src/api/v1/community_businesses/volunteers/__tests__/projects.test.integration.ts]: `/community-businesses/{id}/volunteers` -> `/v1/community-businesses/me/volunteers/projects` ??

File: src/api/v1/community_businesses/volunteers/__tests__/projects.test.integration.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+  });
+
+  describe('GET /community-businesses/me/volunteers/projects', () => {
+    test('can fetch own projects', async () => {
+      const res = await server.inject({
+        method: 'GET',
+        url: '/v1/community-businesses/me/volunteers/projects',
+        credentials: {
+          scope: ['volunteer_logs-own:read'],
+          organisation,
```
Comment (astroash) [inline src/api/v1/community_businesses/volunteers/__tests__/projects.test.integration.ts]: can you add a user so it is clear who these tests is intended for?

File: src/api/v1/community_businesses/volunteers/__tests__/projects.test.integration.ts [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+      const res = await server.inject({
+        method: 'POST',
+        url: '/v1/community-businesses/2/volunteers/projects',
+        credentials: {
+          scope: ['volunteer_logs-own:write'],
+          organisation,
+        },
+      });
+
+      expect(res.statusCode).toBe(404);
```
Comment (astroash) [inline src/api/v1/community_businesses/volunteers/__tests__/projects.test.integration.ts]: why `404` not `401`?

Comment (eliascodes) [reply src/api/v1/community_businesses/volunteers/__tests__/projects.test.integration.ts]: Because the general `{id}` route has not been implemented.

**Pull request merged**